### ISSUE TITLE
Add features from BlockPlanet

### ISSUE THREAD

**Issue by LNJ2** [minetest/minetest#4066]
BlockPlanet is a fork of Minetest by @BlockMen !
It contains many new features like slippery ice, meshes can wave, new croplike drawtype, random waving offsets ...
But BlockMen has left Minetest (and also BlockPlanet), so BlockPlanet will be more and more out of date.
I don't want that these features will be lost, so I created this issue. I would like to see them in Minetest! :)

I don't understand much from the C++ code of Minetest, so I think its better if somebody else could make PRs for these features.

As I saw these features are not unstable, laggy or somewhat.

**Comment by asl97** [comment]
minetest is license under LGPL 2.1 while blockplant is license as [GPL 3](https://github.com/blockplanet/blockplanet/blob/master/LICENSE.txt)

the license ain't compatible.

**Comment by PilzAdam** [comment]
The [first commit](https://github.com/blockplanet/blockplanet/commit/f2722732a98cd63a96f3e104b7c1b04ebeb6e471) @BlockMen made was to change the license so porting code back to Minetest is impossible with our current license.

**Comment by LNJ2** [comment]
So we would have to ask BlockMen, if he changes the license or we change the license from Minetest (what won't happen I think)

**Comment by sfan5** [comment]
We can't just change the license of Minetest just because it fits nicely and we feel like doing that right now. As BlockMen changed the license of blockplanet probably because he <i>didn't</i> want to have changes backported, he will most likely not give permission.

tl;dr <b>no!</b>

**Comment by LNJ2** [comment]
So I just hope that he answers to this issue, even if he don't agrees.

**Comment by est31** [comment]
Yes, unless BlockMen changes the license, none of his changes can be added back to minetest.

Therefore I'd close this issue until that (unlikely) event happens.

**Comment by LNJ2** [comment]
Ok

**Comment by LNJ2** [comment]
BlockMen said this to me in a private email:

> Es gab 2-3 offene PR von mir, als ich Minetest verlassen habe, die Backports von Blockplanet sind. Alle weiteren Änderungen sind und bleiben unter GPL 3.0 lizenziert.

In English:
There were 2-3 open PRs by me, as I left Minetest, that are backports from Blockplanet. All other changes are and remain licensed under GPL 3.0.

**Comment by LNJ2** [comment]
Here are the PRs by BlockMen and I really don't understand why they don't get merged?

https://github.com/minetest/minetest/pulls/BlockMen

**Comment by Zeno-** [comment]
If they have been submitted as PRs by BlockMen then they can potentially be added

**Comment by PilzAdam** [comment]
There are already PRs, why keep this issue open?

**Comment by rubenwardy** [comment]
Any good ideas could be made as separate PRs / issues. If the code isn't compatible, you could clean room implement it, taking only the idea and the wanted behaviour.
